tools/tasks.py

```diff
@@ -407,11 +409,7 @@ def redact_and_zip(self, filename, log_type, salt, node):
         redactor = LogRedactor(salt, self.tmpdir)
 
         for name, fp in self.files.items():
-            if not (
-                ".gz" in name
-                or "expvars.json" in name
-                or os.path.basename(name) == "sync_gateway"
-            ):
+            if not redactable_file(name) or ".gz" in name:
```

Comment:
I know this is covered by unit tests so I know it works, but the precendece here is quite difficult to parse.

Does `not` negate both `or` clauses, or just the first? The brackets made this obvious before, but it's not that easy to determine any more.

The main concern here is that pprof files have the pb.gz suffix so end up matching both.

Would be kinda nice if `redactable_file` could handle this case inside the function?

Reply:
I rewrote this in a kind of aggressive way but I think this is much more robust for anything we add to sgcollect.

The reason that there is this wacky parsing is that text files are opened in latin1. We don't need to figure out the encoding of the files, although I am certain our log files are never latin1. By opening the files in binary mode, we can just do redaction on the bytes.

I created an open function which will detect if it is gzip or not, so it will automatically handle gzipped files or non gzipped files.

This would actually probably mean the pprof files wouldn't get corrupted since almost certainly these regexes wouldn't match. That said, we are now avoiding redacting these files.